### PR TITLE
Cache git repo

### PR DESCRIPTION
**Pull request by darkofabijan**

### CODE REVIEW

File: libcheckout [hunk 1]
```diff
@@ -1,33 +1,117 @@
...
+
+  if [[ ! -z "$cache_age" ]]; then
+    local now=$(date +%s)
+    local diff=$(expr $now - $cache_age)
+    echo "diff: $diff"
+    if (( diff > $SEMAPHORE_GIT_CACHE_AGE )); then
+      echo "Git cache outdated, refreshing..."
+      cd ..
+      cache delete $cache_key
+      cache store git-cache-`date +%s` $SEMAPHORE_GIT_DIR
```
Comment (shiroyasha) [inline libcheckout]: What happens if two jobs try to store to the cache at the same time? Will the storage system collide?

This is not out of the realm of possibility. Parallel jobs are likely candidates to cause this collision at least once in a while.

File: libcheckout [hunk 2]
```diff
@@ -1,33 +1,117 @@
...
+  if [ -z ${SEMAPHORE_GIT_CACHE_AGE} ]; then
+    SEMAPHORE_GIT_CACHE_AGE=259200
+  fi
+
+  local cache_key=$(cache list 2>&1 | grep git-cache- | awk '{ print $1 }')
+  local cache_age=$(echo $cache_key | head -1 | cut -d'-' -f3)
+
+  if [[ ! -z "$cache_age" ]]; then
+    local now=$(date +%s)
+    local diff=$(expr $now - $cache_age)
```
Comment (shiroyasha) [inline libcheckout]: To be 100% fail-safe, we should also check if the cache_age variable is a number. Something like `[[ $cache_age =~ ^-?[0-9]+$ ]]`

File: libcheckout (hunk 1 above)
Comment (CommanderK5) [reply libcheckout]: Tested that, in case of a collision one of the KEYs will not be uploaded.

Comment (CommanderK5) [reply libcheckout]: Added a unique ID to git-cache store - $SEMAPHORE_JOB_ID

**Pull request merged**